### ISSUE TITLE
Some breakages on new M1 ARM machines

### ISSUE THREAD

**Issue by AndrewSB** [tuist/tuist#2082]
On a machine with an M1 chip, tuist has some permission issues. Specifically, I noticed that
1. the installer doesn't work. It fails on the line when we try to `mv` the unzipped tuist into /usr/local/bin

to workaround, I ran the lines of the installer script manually one by one, adding a sudo before the `mv`, which then succeeded.

2. when running tuist the first time, the symbolic link into /usr/local/bin fails, here's the output:

```
No local versions available. Installing the latest version 1.26.0
Downloading version 1.26.0
Installing...
Version 1.26.0 installed
ln: /usr/local/bin/swift-project: Permission denied
```

to fix, I just ran `sudo tuist` instead and everything was fine.

the rest of the tool works great though!

**Comment by pepibumur** [comment]
Thanks for bringing this up @AndrewSB. I don't have a lot of context on what might have changed with M1, but based on your description it seems it is permissions doing system-wide operations. I'll have to investigate it further.

**Comment by steprescott** [comment]
`tuist edit` generates a project whereby `ProjectDescription` can't be imported due to incompatible architecture.

>Module 'ProjectDescription' was created for incompatible target x86_64-apple-macosx10.15: /Users/<USER>/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ProjectDescription-CTCGIW2290LT.swiftmodule

**Comment by pepibumur** [comment]
@AndrewSB & @steprescott we've got a [PR](https://github.com/tuist/tuist/pull/2154) that fixes the issue that you are facing. We'll try to include it in the next release.

**Comment by AndrewSB** [comment]
I just tried re-installing using `bash <(curl -Ls https://install.tuist.io)`, same installer errors. 

I'd leave this issue open to track that, but upto you

**Comment by AndrewSB** [comment]
just tested `tuist edit`, with the latest from https://storage.googleapis.com/tuist-releases/latest/tuistenv.zip. Looks like it's still broken

<img width="571" alt="Screen Shot 2020-12-29 at 4 58 59 PM" src="https://user-images.githubusercontent.com/3814772/103316707-4cd96f00-49f7-11eb-9140-b25424d13ca5.png">

**Comment by dcvz** [comment]
@pepibumur I've verified with the latest version `1.30.0` that `tuist edit` produces a correct project with a valid `ProjectDescription` 🚀 

====

@AndrewSB and for anyone with a new M1 that's facing an issue using the install script with the error:

```bash
mv: rename /tmp/tuistenv/tuistenv to [...]/bin/tuist: Permission denied
ln: [...]/swift-project: Permission denied
```

Many devs normally use Homebrew on their Macs and as part of the installation process, Homebrew changes the ownership of the `/usr/local/bin` directory to you.. the install script for Tuist takes advantage of that and assumes that permission is valid.

However on M1 machines you either have not installed brew yet or have followed the recommendations of the Homebrew team and installed to `/opt/homebrew` instead - so homebrew has not changed the permissions of `/usr/local/bin`. To solve the installation issue you need to run:

```bash
sudo chown -R $(whoami) /usr/local/bin
```

**Comment by AndrewSB** [comment]
as far as I can tell, the `tuist edit`ing behavior is still broken. when trying to build the temporary manifest project, I see this error

```
/Users/asb/Developer/apex-ios/Tuist/ProjectDescriptionHelpers/Project+Templates.swift:1:8: Module 'ProjectDescription' was created for incompatible target x86_64-apple-macosx10.15: /Users/asb/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/ProjectDescription-17MDZ2E74DOKY.swiftmodule
```

**Comment by nhatlee** [comment]
Any update for this issue? I still faced with same error on M1:
```
mv: rename /tmp/tuistenv/tuistenv to /usr/local/bin/tuist: No such file or directory
ln: /usr/local/bin/swift-project: No such file or directory
chmod: /usr/local/bin/tuist: No such file or directory
Tuist installed. Try running 'tuist'
```

**Comment by tiarnann** [comment]
@nhatlee I had a similar issue. I had to manually create the `bin` folder inside `/usr/local/` then the install script worked for me. 

To solve your issue, I think you can do either of these two options:
1. Check if there is an executable named `tuist` inside`/usr/local/bin/tuist/`, move that executable from the `tuist` folder to the `/usr/local/bin/` folder itself.
2. Or add the path `/usr/local/bin/tuist/` to your PATH environment. Here's an example to do that: https://www.architectryan.com/2012/10/02/add-to-the-path-on-mac-os-x-mountain-lion/

**Comment by nhatlee** [comment]
@tiarnann yes, I manually installed successfully Tuist. My problem is that I created the `bin` and `tuist` folder inside `/usr/local/`, so it will lead to cannot find the Tuist command. 
I fixed it by just create a bin folder, then install manually step by step.

**Comment by pepibumur** [comment]
Hey there! I'll look into this tomorrow and get it fixed :)
